ecosystem/sep-0034.md

```diff
@@ -0,0 +1,160 @@
+
+For example, the python implementation may look something like this:
+```python
+import codecs
+from stellar_sdk import Keypair
+from jwcrypto import jwk, jwt
+
+# the client's SIGNING_KEY keypair
+kp = Keypair.random()
+
+# The first two pieces of the JWS
+jws = jwt.JWT(
+    header={"alg": "EdDSA", "typ": "EdDSA", "kid": kp.public_key},
+    claims={
+        "iss": "https://walletsigningserver.com",
+        "sub": "GALWLDCKL5FJTBGRB2KXVHNXDQTE422YD7U6EULR7PSZ2OXPTJ3WHTLU",
+        "jti": "aa77983a-e550-4d90-8cc2-d661d7f0b8f6",
+        "kid": "GCR5WQYXYT4ECBQ3SBALXHICPEVTWKY75XKKZ3ZMF63EXJ5RCWWDO726",
+        "aud": "https://anchorserver.com",
+        "exp": "1597789801",
+        "iat": "1597703375"
+    }   
+)
+
+# The base64-encoded public and secret keys
+# Stellar uses its own representation, so we use the raw bytes
+public_key_encoded = codecs.encode(kp.raw_public_key(), "base64")
+secret_key_encoded = codecs.encode(kp.raw_secret_key(), "base64")
+
+# the JWK used to sign the token
+key = jwk.JWK(
+    kty="OKP",
+    crv="Ed25519",
+    x=public_key_encoded,
+    d=secret_key_encoded,
+)
+jws.make_signed_token(key)
+print(jws.serialize())
+```
+
+The resulting string will look something like this:
+```
+eyJhbGciOiJFZERTQSJ9
+.
+eyJhdWQiOiJodHRwczovL2FuY2hvcnNlcnZlci5jb20iLCJleHAiOiIxNTk3Nzg5ODAxIiwiaWF0IjoiMTU5NzcwMzM3NSIsImlzcyI6Imh0dHBzOi8vd2FsbGV0c2lnbmluZ3NlcnZlci5jb20iLCJqdGkiOiJhYTc3OTgzYS1lNTUwLTRkOTAtOGNjMi1kNjYxZDdmMGI4ZjYiLCJzdWIiOiJHQ0NYS0NIVkVESDZNR1pZSUhER0lGRDVMSEdEM1JDQVo3TkZMSUFZNk5aQ0ZZQlZTNk1RUEZSNyJ9
+.
+xZi9TX_KI9sXA1XJlm_O5t6FLxs0sx9suBf5Kv4SVqL8Glo_zL4EbPZ6SgjeqBd31xnVQ-b5q4rwpiPIz278AQ
+```
+
+The public key used to generate the third part of the JWS can also be used to verify the integrity of the first and second part. This is what the anchor server would do when a wallet makes a request containing the JWS token. 
+
+## Verifying Requests
+Other proposals will specify how the JWS returned by the client's signing server should be sent to the attribution server. The anchor can keep the public keys of known wallet signing servers, but if the client is unknown, the anchor should request the signing server's [stellar.toml](sep-0001.md) and validate that the value found was used to encrypt the JWS.
+
+Wallet signing servers could also rotate their `SIGNING_KEY`. Anchors receiving JWS tokens from these clients may have to update their local copies of the wallets' public keys.
+
+Once verified, the anchor server can use the JWT attributes with confidence that the integrity of the data has been maintained and that the wallet requesting the transaction and passing the JWS belongs to the wallet's signing server.
+
+### Expiration
+Signing servers should select an expiration time for the JWT that is appropriate for the assumptions and risk of the interactions a user can perform with it. A client may own a transaction at the time the JWT is issued but they may not at some point in the future. 
```

Comment:
```suggestion
Wallet servers should select an expiration time for the JWT that is appropriate for the assumptions and risk of the interactions a user can perform with it. A wallet may own a request at the time the JWT is issued but they may not at some point in the future. 
```

Reply:
This makes it seem like the `request` is not an HTTP request, but a general request for a transaction. This should be disambiguated.